### PR TITLE
RI-7632: add components statistics

### PR DESCRIPTION
**Pull request by ArtemHoruzhenko**
# Add Component Status section to RDI Statistics (API v2)

## Summary

Adds a new "Component Status" table section to RDI statistics by fetching data from the pipeline status endpoint alongside metrics.

## Changes

- **api.v2.rdi.client.ts**: Fetch metrics and status in parallel; graceful fallback if status fails
- **metrics-collections.transformer.ts**: New `transformComponentStatus()` function; Component Status section added after Target Connections

## Component Status Table Columns

Status | Name | Type | Version | Errors

## Testing

Added tests for both client and transformer changes, including edge cases and error handling.

New section:
Screenshot 2026-02-03 at 15 08 16

---

> [!NOTE]
> **Medium Risk**
> Moderate risk: changes the v2 statistics aggregation to make an additional API call and alters the returned `sections` shape/order, which could affect consumers if assumptions about section count/indices exist.
> 
> **Overview**
> Adds a new **"Component Status"** table section to RDI API v2 statistics by fetching pipeline `status` alongside `metric-collections` and merging the results.
> 
> `getStatistics()` now calls both endpoints in parallel, treats metrics failures as fatal, and *gracefully omits* the Component Status section if the status call fails. The metrics transformer now optionally accepts status data and inserts the new section after *Target Connections*, with auto-generated columns and joined error strings; tests were added/updated to cover success, fallback, and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 177dbcdad26e06d88b12b002e875c1a4fc0053ca. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts [hunk 1]
```diff
@@ -100,14 +100,24 @@ export class ApiV2RdiClient extends ApiRdiClient {
...
-      );
+      const [metricsResponse, statusResponse] = await Promise.all([
+        this.client.get<GetMetricsCollectionResponse>(
+          RdiUrlV2.GetMetricsCollections(this.selectedPipeline),
+        ),
+        this.client
+          .get<GetStatusResponse>(
+            RdiUrlV2.GetPipelineStatus(this.selectedPipeline),
+          )
+          .catch(() => ({ data: null })), // Graceful fallback if status fails
```
Comment (KrumTy) [inline redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:111]: in what cases does that happen?

Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:111]: temporal connectivity issues... I don't think it should happen but I also don't want to not show entire statistics in case it happens
